core-library/src/main/java/org/silverpeas/core/contribution/content/wysiwyg/service/WysiwygManager.java

```diff
@@ -315,34 +312,16 @@ String suppressLeadingSlashesOrAntislashes(String path) {
 
   /* doubleAntiSlash */
   String doubleAntiSlash(String path) {
-    int i = 0;
-    String res = path;
-    boolean ok = true;
-    while (ok) {
-      int j = i + 1;
-      if ((i < res.length()) && (j < res.length())) {
-        char car1 = res.charAt(i);
-        char car2 = res.charAt(j);
-        if (!((car1 == '\\' && car2 == '\\') || (car1 != '\\' && car2 != '\\'))) {
-          String before = res.substring(0, j);
-          String after = res.substring(j);
-          if (!after.startsWith("\\\\") && !before.endsWith("\\\\")) {
-            res = before + '\\' + after;
-            i++;
-          }
-        }
-      } else {
-        if (i < res.length()) {
-          char car = res.charAt(i);
-          if (car == '\\') {
-            res = res + '\\';
-          }
-        }
-        ok = false;
+    StringBuilder res = new StringBuilder(path);
+    int k = 0;
+    for (int i = 0, j = 1; i < path.length(); i++, j++) {
+      if (path.charAt(i) == '\\' && ((j < path.length() && path.charAt(j) != '\\') &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1067)